APCAppCore/APCAppCore/UI/TabBarControllers/NewsFeed/APCNewsFeedViewController.m

```diff
@@ -0,0 +1,175 @@
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+
+#import "APCNewsFeedViewController.h"
+#import "APCFeedParser.h"
+#import "APCFeedTableViewCell.h"
+#import "APCAppCore.h"
+#import "APCAppDelegate.h"
+
+@interface APCNewsFeedViewController ()
+
+@property (nonatomic, strong) NSArray *posts;
+
+@property (nonatomic, strong) NSDateFormatter *dateFormatter;
+
+@end
+
+@implementation APCNewsFeedViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    
+    self.refreshControl = [UIRefreshControl new];
+    [self.refreshControl addTarget:self action:@selector(refreshFeed) forControlEvents:UIControlEventValueChanged];
+    
+    self.dateFormatter = [NSDateFormatter new];
+    self.dateFormatter.dateFormat = @"MM/dd/yy";
+    
+    self.title = NSLocalizedString(@"News Feed", nil);
+}
+
+- (void)viewDidAppear:(BOOL)animated
+{
+    [super viewDidAppear:animated];
+    
+    self.posts = [self newsFeedManager].feedPosts;
+    
+    if (self.posts.count > 0) {
+        self.tableView.backgroundView = nil;
+    } else {
+        UILabel *emptyLabel = [UILabel new];
+        emptyLabel.frame = self.tableView.bounds;
+        
+        emptyLabel.text = NSLocalizedString(@"There's nothing here yet.", nil);
+        emptyLabel.textColor = [UIColor appSecondaryColor3];
+        emptyLabel.textAlignment = NSTextAlignmentCenter;
+        emptyLabel.font = [UIFont appMediumFontWithSize:22];
+        
+        self.tableView.backgroundView = emptyLabel;
+    }
+    
+    [self.tableView reloadSections:[NSIndexSet indexSetWithIndex:0] withRowAnimation:UITableViewRowAnimationAutomatic];
+    
+    NSUInteger unreadCount = [[self newsFeedManager] unreadPostsCount];
+    NSNumber *unreadValue = @(unreadCount);
+    
+    APCAppDelegate *appDelegate = (APCAppDelegate *)[[UIApplication sharedApplication] delegate];
+    
+    UITabBarItem *newsFeedTab = appDelegate.tabBarController.tabBar.selectedItem;
+    if (newsFeedTab.tag == (NSInteger)kAPCNewsFeedTabIndex) {
+        if (unreadCount != 0) {
+            newsFeedTab.badgeValue = [unreadValue stringValue];
+        } else {
+            newsFeedTab.badgeValue = nil;
+        }
+    }
+}
+
```

Comment:
I have seen this badgeValue update code a few times, better to have a method and reuse.